services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -1423,16 +1431,31 @@ class ApplicationServiceITest : DatabaseTest() {
         }
 
         @Test
-        fun `when application not in Allu should delete application`() {
+        fun `when application not in Allu should delete application and all its attachments`() {
             val hanke = hankeFactory.saveMinimal()
             val application =
                 applicationFactory.saveApplicationEntity(USERNAME, hanke = hanke, alluId = null)
+            // assert that application is saved to database
             assertThat(applicationRepository.findAll()).hasSize(1)
+            val attachment1 = attachmentFactory.saveAttachment(application.id!!)
+            val attachment2 = attachmentFactory.saveAttachment(application.id!!)
+            // assert that attachments are saved to cloud
+            assertTrue(applicationAttachmentContentService.exists(attachment1.blobLocation!!))
+            assertTrue(applicationAttachmentContentService.exists(attachment2.blobLocation!!))
+            // assert that attachment metadata is saved to database
+            assertThat(applicationAttachmentRepository.findByApplicationId(application.id!!))
+                .hasSize(2)
 
             applicationService.delete(application.id!!, USERNAME)
 
+            // assert that application is deleted from database
             assertThat(applicationRepository.findAll()).isEmpty()
             verify { cableReportServiceAllu wasNot Called }
+            // assert that attachments are deleted from cloud
+            assertFalse(applicationAttachmentContentService.existsForApplication(application.id!!))
```

Comment:
If there are no attachments, there are none for the application either.
```suggestion
            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
```

Reply:
Added `list` method to FileClient:
```
fun list(container: Container, prefix: String): List<String>
```

Reply:
If there are no attachments, there are none for the application. So the prefix isn't needed, at least not here.

We don't want to add test code to production whenever we can avoid it. Adding the method to FileClient means that BlobFileClient will have the same method, but it's not used anywhere. If the method is needed, it could be added to MockFileClient, so it doesn't cause changes on the production side. The test class can inject MockFileClient instead of FileClient since we know it's going to get a MockFileClient anyway.

Reply:
Good point. Thank you!